test/jdk/java/lang/System/i18nEnvArg.java

```diff
@@ -0,0 +1,155 @@
+            if (!dcs.equals(cs)) {
+                return;
+            }
+            ProcessBuilder pb = new ProcessBuilder(javeExe,
+                "--add-opens=java.base/java.lang=ALL-UNNAMED",
+                "-classpath",
+                System.getProperty("java.class.path"),
+                "i18nEnvArg$Verify",
+                text);
+            pb.redirectErrorStream(true);
+            Map<String, String> environ = pb.environment();
+            environ.clear();
+            environ.put("LANG", "ja_JP.eucjp");
+            environ.put(text, text);
+            Process p = pb.start();
+            InputStream is = p.getInputStream();
+            byte[] ba = new byte[maxSize];
+            ByteBuffer bb = ByteBuffer.wrap(ba);
+            int ch;
+            while((ch = is.read()) != -1) {
+                bb.put((byte)ch);
+            }
+            int rc = p.waitFor();
+            System.out.write(ba, 0, bb.position());
+        }
+    }
+
+    public static class Verify {
+        public static void main(String[] args) throws Exception {
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            byte[] kanji = (text + "=" + text).getBytes(cs);
+            String s = System.getenv(text);
+            if (!text.equals(s)) {
+                throw new Exception("getenv() returns unexpected data");
+            }
+            if (!text.equals(args[0])) {
+                System.out.print("Unexpected argument was received: ");
+                for(char ch : text.toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.print("<->");
+                for(char ch : args[0].toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.println();
+            }
+            Class<?> cls = Class.forName("java.lang.ProcessEnvironment");
+            Method enviorn_mid = cls.getDeclaredMethod("environ");
+            enviorn_mid.setAccessible(true);
+            byte[][] environ = (byte[][]) enviorn_mid.invoke(null,
+                (Object[])null);
+            byte[] ba = new byte[maxSize];
+            StringJoiner sj = new StringJoiner(", ");
+            for(int i = 0; i < environ.length; i += 2) {
+                ByteBuffer bb = ByteBuffer.wrap(ba);
+                bb.put(environ[i]);
+                bb.put((byte)'=');
+                bb.put(environ[i+1]);
+                byte[] envb = Arrays.copyOf(ba, bb.position());
+                if (Arrays.equals(kanji, envb)) return;
+                StringBuilder sb = new StringBuilder();
+                for(byte b : envb) {
+                    if (b == 0x5C) {
+                        sb.append("\\x5C");
+                    } else if (b >= 0x20 && b <= 0x7F) {
+                        sb.append((char)b);
+                    } else {
+                        sb.append(String.format("\\x%02X", (int)b & 0xFF));
```

Comment:
The methods of java.util.HexFormat may be useful here.
It seems that the "5C" would fit into this "else" clause and not need a separate statement.

Reply:
Use HexFormat class